### ISSUE TITLE
No locking on script_executions.json

### ISSUE THREAD

**Issue by wittwerch** [mautic/mautic#1576]
## Description
Executing multiple Mautic command over cronjob on the same time leads to race conditions. I'm using Mautic 1.3.1

```
#Ansible: foo: Update campaigns
*/5 * * * * /var/www/foo/app/app/console --quiet mautic:campaigns:update
#Ansible: foo: Trigger campaigns
*/5 * * * * /var/www/foo/app/app/console --quiet mautic:campaigns:trigger
#Ansible: foo: Update lead lists
*/1 * * * * /var/www/foo/app/app/console --quiet mautic:leadlists:update
```

It looks like the entries in script_executions.json are not cleared properly. The job is not running anymore, I'v checked that.
```
{  
   "in_progress":{  
      "mautic:leadlists:update":{  
         "all":1460010902
      },
      "mautic:campaigns:update":{  
         "all":1460010902
      },
      "mautic:campaigns:trigger":[  

      ]
   }
}
```

So from that moment on, execution is blocked because.
```
root@web002:~# /var/www/foo/app/app/console mautic:leadlists:update
Script in progress. Use -f or --force to force execution.
```
I had a look at the code and I guess the problem are some race conditions on the script_executions.json file. Reading and writing the file is done without a lock, so it's possible that two process write to the file and overwrite each other.

## Steps to reproduce
Execute multiple commands at the same time.

**Comment by escopecz** [comment]
This issue seems to be related: https://github.com/mautic/mautic/issues/1664 and possibly contains also a fix. @wittwerch could you check closely?

**Comment by jkuchar** [comment]
Thanks for posting this, I'm having same problem with my instance.

**Comment by maxitromer** [comment]
I'm writing because I was the same problem and I can fix it using the solution proposed by **@ioncube** in #1664.

**Comment by wittwerch** [comment]
Yes, #1664 fixes the problem!

**Comment by PatchRanger** [comment]
I guess we should close this as a duplicate of https://github.com/mautic/mautic/issues/1664.

**Comment by alanhartless** [comment]
Closing this as it was fixed with #1867.
